llama-index-integrations/vector_stores/llama-index-vector-stores-pinecone/llama_index/vector_stores/pinecone/base.py

```diff
@@ -433,7 +433,7 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
                     "values": [v * (1 - query.alpha) for v in sparse_vector["values"]],
                 }
 
-        query_embedding = None
+        query_embedding = [0.0] * len(query.query_embedding)
```

Comment:
This is specifically to fix the case where query_mode is sparse right?

Won't providing a vector of zeros still kind of influence the search result? If we are in sparse mode, should we just query without the vector kwarg? Or should we set the alpha to completely ignore the Dense zero vector?

Reply:
yup

The issue is that if we query without the vector kwarg, we get an error when making the request. In their hybrid search tutorial, pinecone says we should scale the dense vector by alpha (in this case zero) before querying, so using a vector of zeros in this case should be fine (it's overwritten later for other query modes anyways) https://www.pinecone.io/learn/hybrid-search-intro/

Reply:
Ah great, that works for me!

Reply:
Ok wait, one more worry haha -- we should ensure that the query embedding is not none before doing this

Reply:
true, added a check